tester/src/test/scala/spinal/tester/scalatest/SpinalSimRegIfTester.scala

```diff
@@ -0,0 +1,233 @@
+    case AccessType.WO => null
+    case AccessType.WOC => null
+    case AccessType.WOS => null
+    case AccessType.W1 => curState
+    // 24 - 27
+    case AccessType.WO1 => null
+    case AccessType.NA => curState
+    case AccessType.W1P => curState
+    case AccessType.W0P => curState
+
+    case _ => null
+  }
+
+  // Result is the NEXT state of the register
+  private def onWrite(accType: AccessType, curState: BigInt, in: BigInt, firstWrite: Boolean): BigInt = accType match {
+    // 0 - 3
+    case AccessType.RO => curState
+    case AccessType.RW => in
+    case AccessType.RC => curState
+    case AccessType.RS => curState
+    // 4 - 7
+    case AccessType.WRC => in
+    case AccessType.WRS => in
+    case AccessType.WC => 0
+    case AccessType.WS => (BigInt(1) << bitWidth) - 1
+    // 8 - 11
+    case AccessType.WSRC => (BigInt(1) << bitWidth) - 1
+    case AccessType.WCRS => 0
+    case AccessType.W1C => (((BigInt(1) << bitWidth) - 1) ^ in) & curState
+    case AccessType.W1S => in | curState
+    // 12 - 15
+    case AccessType.W1T => in ^ curState
+    case AccessType.W0C => in & curState
+    case AccessType.W0S => (((BigInt(1) << bitWidth) - 1) ^ in) | curState
+    case AccessType.W0T => (((BigInt(1) << bitWidth) - 1) ^ in) ^ curState
+    // 16 - 19
+    case AccessType.W1SRC => in | curState
+    case AccessType.W1CRS => (((BigInt(1) << bitWidth) - 1) ^ in) & curState
+    case AccessType.W0SRC => (((BigInt(1) << bitWidth) - 1) ^ in) | curState
+    case AccessType.W0CRS => in & curState
+    // 20 - 23
+    case AccessType.WO => in
+    case AccessType.WOC => 0
+    case AccessType.WOS => (BigInt(1) << bitWidth) - 1
+    case AccessType.W1 => if (firstWrite) in else curState
+    // 24 - 27
+    case AccessType.WO1 => if (firstWrite) in else curState
+    case AccessType.NA => in
+    case AccessType.W1P => in
+    case AccessType.W0P => in
+
+    case _ => curState
+  }
+
+  test("random") {
+    SimConfig.compile(new RegIfTester()).doSim { dut =>
+      dut.clockDomain.forkStimulus(10 ns)
+
+      dut.io.t #= false
+      dut.io.i.randomize()
+
+      val driver = AxiLite4Driver(dut.io.bus, dut.clockDomain)
+      driver.reset()
+
+      val writeHistory = mutable.HashSet[BigInt]()
+
+      val readMonitor = new AxiLite4ReadOnlyMonitor(dut.io.bus.ar, dut.io.bus.r, dut.clockDomain) {
+
+        var beatData: BigInt = 0
+
+        override def onReadStart(addr: BigInt): Unit = {
+          beatData = 0
+        }
+
+        override def onReadByte(addr: BigInt, data: Byte): Unit = {
+          val addrAlign = addr >> log2Up(busConfig.bytePerWord) << log2Up(busConfig.bytePerWord)
+          var biData = BigInt(data)
+          if (biData < 0)
+            biData = biData + 256
+          beatData |= biData << ((addr - addrAlign) * 8).toInt
+        }
+
+        override def onResponse(addr: BigInt, resp: Byte): Unit = {
+          val regData = (dut.io.o.toBigInt >> (addr*8).toInt) & ((BigInt(1) << busConfig.dataWidth) - 1)
```

Comment:
How about adding wordRange and symbolRange to AxiLite4Config, such as what [Axi4Config](https://github.com/SpinalHDL/SpinalHDL/blob/e5a7dc970b498de633c0301a7fbcecceeb3ba669/lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4.scala#L89) do?

Reply:
Not sure about this suggestions.  This particular code slices out the effected bits from the wide register for verification.

Reply:
What I mean is to use wordRange to get better understandable (might) code. :)
``` scala
val addrAlign = addr >> log2Up(busConfig.bytePerWord) << log2Up(busConfig.bytePerWord)
val addrAlign = addr(busConfig.wordRange).resizeLeft(busConfig.addressWidth) // did not tested.
```

Reply:
Ah but won't that generate hardware in the sim thread?

Reply:
Yes, that is a problem. 😆

Reply:
How bad is it?

```scala
.compile(new Component extends MyDut {
  val additionalSignalsHere = basedOnDutOutputs
})
```

or

```scala
.compile(new Component {
  val internals = new MyDut
  val io = internals.io
  val additionalSignalsHere = basedOnDutOutputs
})
```

XD

What if SpinalHDL had High-Level Simulation (HLS?… HLSim!) features? XD For comb logic it would require to re-implement all operators and functions in software world for base types and then turn these not-created signals into objects where readers (`.toBigInt`) evaluate the logic instead of querying the simulator. Then for seq logic… *oh no*